src/java.desktop/share/classes/java/awt/Window.java

```diff
@@ -3865,7 +3861,7 @@ private static void setLayersOpaque(Component component, boolean isOpaque) {
             if (content != null) {
                 content.setOpaque(isOpaque);
 
-                // Iterate down one level to see whether we have a JApplet
+                // Iterate down one level to see whether we have (eg) a JInternalFrame
```

Comment:
I think it is more related to top level windows like JWindow/JDialog etc.

Reply:
After the applet removal, this could be dead code. It might be better not to mention JInternalFrame unless we have prove that the code actually works for it, add a TODO? or something?

Reply:
it could be related to appletviewer